docs/design/queue-guarantee-resource-reservation-design.md

```diff
@@ -0,0 +1,65 @@
+# Volcano Resource Reservation For Queue
+
+@[qiankunli](https://github.com/qiankunli); Oct 11rd, 2021
+
+## Motivation
+As [issue 1101](https://github.com/volcano-sh/volcano/issues/1101) mentioned, Volcano should support resource reservation
```

Comment:
Thanks for your design doc. Please add the following parts and restruct the doc.
1. Scope 
2. Out of Scope ( clarify the parts which you won't implement in this project. f.g. enhance vcctl/vqueue to configure gurantee)
3. Error handing
4. Features Interaction
The gurantee has relation with capability and proporiton behavior or even confliction with them.  you can write down the degin behavior in these use cases or scenarios.
let's say:
## use case1:   
The proportion is enabled and then create a new queue with a big gurantee value that can not be satisfied. what's the design behavor against it. Reject the creation or preempt other job or just let the job in this new queue waiting for releasing resource.

you can refer to this doc:
https://github.com/volcano-sh/volcano/blob/master/docs/design/job-api.md